pytest_catchlog.py

```diff
@@ -92,19 +97,93 @@ def pytest_addoption(parser):
+                   dest='log_format',
+                   default=DEFAULT_LOG_FORMAT,
                    help='log format as used by the logging module.')
     add_option_ini(parser,
                    '--log-date-format',
-                   dest='log_date_format', default=None,
+                   dest='log_date_format',
+                   default=DEFAULT_DATE_FORMAT,
                    help='log date format as used by the logging module.')
+    add_option_ini(parser,
+                   '--log-extra-level',
+                   dest='log_extra_levels',
+                   type='args',
+                   action='append',
+                   help='extra logging levels that verbosity should be sensible to.')
 
 
 def pytest_configure(config):
     """Always register the log catcher plugin with py.test or tests can't
     find the  fixture function.
     """
+    verbosity = config.getoption('-v')
+    if verbosity <= 1:
+        cli_handler_level = logging.FATAL
+    elif verbosity > 1:
+        # Prepare the handled_levels dictionary
+        log_levels = []
+        available_levels = set([
+            #logging.FATAL,  # This will be the default console level if no -v is passed
+            logging.CRITICAL,
+            logging.ERROR,
+            logging.WARNING,
+            logging.INFO,
+            logging.DEBUG,
+            logging.NOTSET
+        ])
+        for level in get_option_ini(config, 'log_extra_levels'):
+            try:
+                level_num = int(getattr(logging, level, level))
+            except ValueError:
+                # Python logging does not recognise this as a logging level
+                raise pytest.UsageError(
+                    "'{0}' is not recognized as a logging level name. Please "
+                    "consider passing the logging level num instead.".format(
+                        level))
+            available_levels.add(level_num)
+
+        for level in available_levels:
+            if level > logging.CRITICAL:
+                # -v set's the console handler logging level to CRITICAL,
+                # higher log level messages, ie, >= FATAL are always shown
+                # because that's the default level set for the handler
+                continue
+            if level < logging.NOTSET:
+                # Log levels lower than NOTSET, we're not interested
+                continue
+            if level in log_levels:
+                # We already know about this log level
+                continue
+            log_levels.append(level)
+
+        # Build a dictionary mapping of verbosity level to logging level
+        # verbosity=0         FATAL    (no pytest output is shown and FATAL
+        #                               log messages are displayd)
+        # verbosity=1   -v    CRITICAL (pytest verbosity kicks in, but only
+        #                               log messages with higher or equal
+        #                               level to CRITICAL are shown)
```

Comment:
But why is the highest level not included at the first place? It would eliminate the need to start counting from 1 in the `enumerate` call below, and let us use zero-based indices I was talking about yesterday...